File: data/readable_format/cpp/file274.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

class SpeciesNetwork {
private:
    static const int max_species = 10;
    vector<string> species_names;
    vector<int> atomic_numbers;
    int num_species;

public:
    SpeciesNetwork() : num_species(0) {
        species_names.resize(max_species);
        atomic_numbers.resize(max_species);
    }

    void init_network() {
        species_names[0] = "Hydrogen";
        atomic_numbers[0] = 1;
        species_names[1] = "Helium";
        atomic_numbers[1] = 2;
        num_species = 2;
    }

    int find_species_index(const string& name) {
        for (int i = 0; i < num_species; ++i) {
            if (species_names[i] == name) {
                return i + 1; // Adjusting index to match Fortran's 1-based indexing
            }
        }
        return 0; // If not found, return 0
    }
};

void test_species_network() {
    SpeciesNetwork network;
    network.init_network();

    cout << "Testing SpeciesNetwork:" << endl;
    cout << "Index of 'Hydrogen': " << network.find_species_index("Hydrogen") << endl;
    cout << "Index of 'Helium': " << network.find_species_index("Helium") << endl;
    cout << "Index of 'Oxygen' (expected to be 0): " << network.find_species_index("Oxygen") << endl;
}

int main() {
    test_species_network();
    return 0;
}

File: data/readable_format/fortran/file274.f90
program species_network_test
    implicit none
    
    ! Declarations
    integer, parameter :: max_species = 10
    character(len=10), dimension(max_species) :: species_names
    integer, dimension(max_species) :: atomic_numbers
    integer :: num_species = 0
    integer :: i

    ! Initialize the species network with some example data
    call init_network()

    ! Test: Output the indices of some species
    write(*,*) 'Index of Hydrogen:', find_species_index('Hydrogen')
    write(*,*) 'Index of Helium:', find_species_index('Helium')
    write(*,*) 'Index of Oxygen (expected to be 0):', find_species_index('Oxygen')

contains

    subroutine init_network()
        species_names(1) = 'Hydrogen'
        atomic_numbers(1) = 1
        species_names(2) = 'Helium'
        atomic_numbers(2) = 2
        num_species = 2
    end subroutine init_network
    
    function find_species_index(name) result(index)
        character(len=*), intent(in) :: name
        integer :: index
        integer :: i

        index = 0
        do i = 1, num_species
            if (trim(species_names(i)) == trim(name)) then
                index = i
                exit
            endif
        end do
    end function find_species_index

end program species_network_test